sdks/java/io/amazon-web-services/src/main/java/org/apache/beam/sdk/io/aws/sns/SnsIO.java

```diff
@@ -256,9 +269,36 @@ public Write withResultOutputTag(TupleTag<PublishResult> results) {
       return builder().setResultOutputTag(results).build();
     }
 
+    /**
+     * Encode the full {@code PublishResult} object, including HTTP status code and response
+     * headers.
+     */
+    public Write withFullPublishResult() {
+      return builder()
+          .setCoderProvider(
+              CoderProviders.forCoder(
+                  TypeDescriptor.of(PublishResult.class), FullPublishResultCoder.of()))
+          .build();
+    }
+
+    /**
+     * Encode the minimal {@code PublishResult} object, including HTTP status code but excluding
+     * HTTP response headers.
```

Comment:
You might want to mention that this is the default here.

Reply:
Actually I left the default behavior the same for backward compatibility, i.e. encode only the messageId using (what is now called) DefaultPublishResultCoder. This method selects the MinimalPublishResultCoder which encodes messageId, requestId, and statusCode but not the HTTP headers.